src/Compilers/CSharp/Test/Semantic/Semantics/CollectionLiteralTests.cs

```diff
@@ -1034,1026 +1290,2634 @@ static void Main()
+                        IL_0017:  ldloc.2
+                        IL_0018:  callvirt   "void System.Collections.Generic.List<int>.Add(int)"
+                        IL_001d:  ldloc.1
+                        IL_001e:  callvirt   "bool System.Collections.IEnumerator.MoveNext()"
+                        IL_0023:  brtrue.s   IL_000f
+                        IL_0025:  leave.s    IL_0031
+                      }
+                      finally
+                      {
+                        IL_0027:  ldloc.1
+                        IL_0028:  brfalse.s  IL_0030
+                        IL_002a:  ldloc.1
+                        IL_002b:  callvirt   "void System.IDisposable.Dispose()"
+                        IL_0030:  endfinally
+                      }
+                      IL_0031:  ldloc.0
+                      IL_0032:  callvirt   "int[] System.Collections.Generic.List<int>.ToArray()"
+                      IL_0037:  ret
+                    }
+                    """,
+                ("int[]", "int[]") =>
+                    // PROTOTYPE: Shouldn't require an intermediate List<int> since the compiler can use e.Length directly.
+                    """
+                    {
+                      // Code size       40 (0x28)
+                      .maxstack  2
+                      .locals init (System.Collections.Generic.List<int> V_0,
+                                    int[] V_1,
+                                    int V_2,
+                                    int V_3)
+                      IL_0000:  newobj     "System.Collections.Generic.List<int>..ctor()"
+                      IL_0005:  stloc.0
+                      IL_0006:  ldarg.0
+                      IL_0007:  stloc.1
+                      IL_0008:  ldc.i4.0
+                      IL_0009:  stloc.2
+                      IL_000a:  br.s       IL_001b
+                      IL_000c:  ldloc.1
+                      IL_000d:  ldloc.2
+                      IL_000e:  ldelem.i4
+                      IL_000f:  stloc.3
+                      IL_0010:  ldloc.0
+                      IL_0011:  ldloc.3
+                      IL_0012:  callvirt   "void System.Collections.Generic.List<int>.Add(int)"
+                      IL_0017:  ldloc.2
+                      IL_0018:  ldc.i4.1
+                      IL_0019:  add
+                      IL_001a:  stloc.2
+                      IL_001b:  ldloc.2
+                      IL_001c:  ldloc.1
+                      IL_001d:  ldlen
+                      IL_001e:  conv.i4
+                      IL_001f:  blt.s      IL_000c
+                      IL_0021:  ldloc.0
+                      IL_0022:  callvirt   "int[] System.Collections.Generic.List<int>.ToArray()"
+                      IL_0027:  ret
+                    }
+                    """,
+                ("ReadOnlySpan<int>", "ReadOnlySpan<int>") =>
+                    // PROTOTYPE: Shouldn't require an intermediate List<int> since the compiler can use e.Length directly.
```

Comment:
it might be worth demonstrating with both `ReadOnlySpan<int>` and `scoped ReadOnlySpan<int>` parameters. It feels like the compiler has the option to make the non-scoped case a *no-op* but not the scoped case.